website_sale_payment_term/models/sale_order.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.model
+    def create(self, vals):
+        vals = self._set_payment_term_id(vals)
+        return super(SaleOrder, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        vals = self._set_payment_term_id(vals)
+        return super(SaleOrder, self).write(vals)
+
+    def _set_payment_term_id(self, vals):
+        payment_acquirer_id = vals.get('payment_acquirer_id')
+        payment_term_id = self.env['payment.acquirer'].browse(
```

Comment:
Will fail if there is no payment_acquirer_id in vals.

Just start with:
```
if vals.get('payment_term_id') or not vals.get('payment_acquirer_id'):
    return vals
```
Then you know you have a payment acquirer and are not overwriting an explicitly requested payment_term_id.

Reply:
I am not sure how this will fail.

If there is no `payment_acquirer_id` in vals then that whole `browse(None).field` will evaluate to `False`. And then the condition will evaluate to `False` again.

Reply:
True it does not fail, did not expect browse(False) to just return an empty record set. Still an unneeded and easily avoidable function call...